src/test/java/io/event/thinking/eventstore/sample/SubscriptionCommandHandler.java

```diff
@@ -0,0 +1,65 @@
+package io.event.thinking.eventstore.sample;
+
+import io.event.thinking.eventstore.Criteria;
+import io.event.thinking.eventstore.Event;
+import io.event.thinking.eventstore.EventStore;
+import io.event.thinking.eventstore.SequencedEvent;
+import io.event.thinking.eventstore.Tag;
+import reactor.core.publisher.Mono;
+
+import static io.event.thinking.eventstore.ConsistencyCondition.consistencyCondition;
+import static io.event.thinking.eventstore.Criteria.criteria;
+import static io.event.thinking.eventstore.Criterion.criterion;
+import static io.event.thinking.eventstore.Tag.tag;
+import static io.event.thinking.eventstore.sample.Constants.*;
+
+class SubscriptionCommandHandler {
+
+    private final EventStore eventStore;
+
+    SubscriptionCommandHandler(EventStore eventStore) {
+        this.eventStore = eventStore;
+    }
+
+    Mono<Long> handle(SubscribeStudent cmd) {
+        var criteria = buildCriteria(cmd);
+        // source events
+        var result = eventStore.read(criteria);
+        return result.flux()
+                     .map(SequencedEvent::event)
+                     // build the command model based on sourced events
+                     .reduce(new Subscription(), this::applyEvent)
+                     // execute the command
+                     .flatMap(Subscription::subscribe)
```

Comment:
You are right! Execution is validation + event publication. Maybe a better name for the method would be `validateSubscription`? I was on the verge of moving the `Subscription` command model to the command handler since `Subscription` feels a bit anemic. However, I wanted to remove any `event store` references from the `Subscription`.

Reply:
Would returning events to the caller be a possible solution?

Reply:
That might be the cleanest solution, after all!